lib/hooks/useAuth.ts

```diff
@@ -157,47 +159,54 @@ export const useAuth = () => {
       } else {
         setCognitoError(t("InternalServiceException"));
       }
-    } finally {
-      setSubmitting(false);
     }
+    // set the formik submitting state to false
+    setSubmitting(false);
 
-    // Automated sign in with provided credentials
+    // end the execution of the function if the confirmation did not succeed
     if (!confirmationSuccess) return;
-    try {
-      const response = await signIn<"credentials">("credentials", {
-        redirect: false,
-        username,
-        password,
-      });
 
-      if (response?.error) {
-        const responseErrorMessage = response.error;
-        logMessage.error(responseErrorMessage);
-        if (
-          responseErrorMessage.includes("UserNotFoundException") ||
-          responseErrorMessage.includes("NotAuthorizedException")
-        ) {
-          confirmationAuthenticationFailedCallback(
-            t("UsernameOrPasswordIncorrect.title"),
-            t("UsernameOrPasswordIncorrect.description"),
-            t("UsernameOrPasswordIncorrect.link"),
-            t("UsernameOrPasswordIncorrect.linkText"),
-            false
-          );
-        } else if (responseErrorMessage.includes("GoogleCredentialsExist")) {
-          await router.push("/admin/login");
-        } else {
-          setCognitoError(t("InternalServiceException"));
+    // try and sign the user in automatically if shouldSignIn is true, otherwise just
+    // call the passed in confirmationCallback
+    if (shouldSignIn) {
```

Comment:
Maybe use an early return here to get rid of some nesting of ifs

if (!shouldSignIn) {
confirmationCallback();
return 
}

Reply:
Yup can do